moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h

```diff
@@ -50,28 +50,28 @@ namespace moveit
 {
 namespace planning_interface
 {
-MOVEIT_CLASS_FORWARD(MoveItCpp);
+MOVEIT_CLASS_FORWARD(MoveItCpp)
 
 class MoveItCpp
 {
 public:
   /// Specification of options to use when constructing the MoveItCpp class
   struct PlanningSceneMonitorOptions
   {
-    void load(const ros::NodeHandle& nh)
+    void load(const rclcpp::Node::SharedPtr& node)
```

Comment:
Any reason to not use the templated approach that is popular in the ros2 code base? 

```   
template<class NodeT>
void load(NodeT && node){
    return load(node->get_node_parameters_interface())
}
void load(rclcpp::node_interfaces::NodeParametersInterface::SharedPtr& node_parameters_interface){
    ...
}
```

Reply:
Right now this is just a migration step so I didn't consider it. We're planning to make a bigger refactoring for parameter loading that includes declaring all parameters that we use. However, I don't see a real upside to use the template version, why do you think we should use it?

Reply:
It would be nice to be able to use weak pointers but to be honest, I really don't see that as a strong enough use case for going back and templating everything

Reply:
Ok, I see that it can make sense of you structure your library around the node components (which would actually be very nice) but introducing this in an existing library is probably not that easy. I would be interested in some discussions how we can improve node management in MoveIt for later milestones.